internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -465,7 +465,7 @@ func convertEnvSecurityGroupCfg(mft *manifest.Environment) (*template.SecurityGr
 
 func (s *LoadBalancedWebService) convertNetworkLoadBalancer() (networkLoadBalancerConfig, error) {
 	nlbConfig := s.manifest.NLBConfig
-	if nlbConfig.IsEmpty() {
```

Comment:
I think we should implement `IsEmpty` for `manifest.NLB` as well, and calls `nlbConfg.IsEmpty()` here, instead if `nlbConfig.MainListener.IsEmpty()`.  

It is true that for a valid NLB config, `nlbConfig.MainListener.IsEmpty()` implies that  `nlbConfig.IsEmpty()`, but I think this is an assumption that client package should not be aware of, but rather hidden within `manifest` package. LMK if this makes sense!

Reply:
Yeah I think it does a lot of sense. I totally agree that client shouldn't make any assumption that `nlbConfig.MainListener.IsEmpty()` implies that `nlbConfig.IsEmpty()`. I have added this method which is really useful for validations as well.